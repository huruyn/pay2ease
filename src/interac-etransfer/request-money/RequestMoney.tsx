import { useState, useEffect } from 'react';
import {
  Alert,
  Col,
  Row,
} from 'react-bootstrap';
import { useNavigate, useParams } from 'react-router-dom';
import { useMsal } from '@azure/msal-react';
import StepComponent from '../../common/StepComponent';
import RequestDetails from './components/RequestDetails';
import RequestSent from './components/RequestSent';
import RequestMoneyVerificationModal from '../dialogs/RequestMoneyVerificationModal';
import Api, { Account, Contact, InteracEtransferTransaction } from '../../api';
import { TransferDetails } from '../dialogs/SendMoneyVerificationModal';
import CommonPageContainer from '../../common/CommonPageContainer';

export const enum PageIds {
  DetailPageId = 'request-detail',
  VerifyPageId = 'request-verify',
  SentPageId = 'request-sent',
}

export interface RequestMainDetails {
  amount: number;
  invoiceNumber: number;
  message: string;
  notifyByEmail: boolean;
  notifyTextMessage: boolean;
  agreed: boolean;
}

const Divider = () => <div className="border-top my-3" />;

const StepIndexes: any = {
  'request-detail': 1,
  'request-verify': 1,
  'request-sent': 2,
};

export default function RequestMoney() {
  const { stepId = 'request-detail' } = useParams();

  const navigate = useNavigate();
  const [currentStep, setCurrentStep] = useState(StepIndexes[stepId || 'request-detail']);

  const [isRequestingMoney, setIsRequestingMoney] = useState(false);
  const [showVerifyModal, setShowVerifyModal] = useState(false);
  const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
  const [selectedContact, setSelectedContact] = useState(0);
  const [selectedAccount, setSelectedAccount] = useState(0);

  const [mainInfo, setMainInfo] = useState<RequestMainDetails>({
    amount: 0,
    message: '',
    invoiceNumber: 0,
    notifyByEmail: false,
    notifyTextMessage: false,
    agreed: false,
  });

  const [accountsList, setAccountsList] = useState<Account[] | null>([]);
  const [contactList, setContactList] = useState<Contact[] | null>([]);
  const { instance, accounts } = useMsal();

  const validateInputs = (): string | null => {
    if (selectedContact === 0) return 'Please select a contact to request the money from';
    if (selectedAccount === 0) return 'Please select an account';
    if (mainInfo.amount <= 0) return 'Amount should be greater than 0';
    if (mainInfo.amount > 3000) return 'The maximum amount you can send in each transfer is $3,000';
    if (!mainInfo.agreed) return 'Please confirm that you have existing relationship with this contact';
    return null;
  };

  const navigateStep = (pageId: PageIds) => {
    const validationMessage = validateInputs();
    if (validationMessage) {
      setErrorMessage(validationMessage);
      return;
    }
    navigate(`/interac-etransfer/request-money/${pageId}`);
  };

  const getTransferDetails = (): TransferDetails => {
    const sourceAccount: Account = accountsList!.find(
      (x) => x.id === selectedAccount,
    )!;
    const destinationContact: Contact = contactList!.find(
      (x) => x.id === selectedContact,
    )!;
    return {
      amount: mainInfo.amount,
      source: {
        name: sourceAccount?.name ?? '',
      },
      destination: {
        firstName: destinationContact?.firstName,
        lastName: destinationContact?.lastName,
        email: destinationContact?.email ?? '',
      },
      fromAccount: sourceAccount?.name,
    };
  };

  const handleRequestMoneyVerificationClose = () => {
    navigateStep(PageIds.DetailPageId);
  };

  const handleRequestMoneyVerificationBack = () => {
    setCurrentStep(1);
    navigateStep(PageIds.DetailPageId);
  };

  useEffect(() => {
    const currentApi = new Api(instance, accounts[0]);

    (async () => {
      try {
        const result = await currentApi.listAccounts();
        setAccountsList(result);
      } catch (err) {
        setTimeout(() => {
          setErrorMessage('Sorry! a problem has occurred when getting accounts.');
        }, 0);
      }

      try {
        const result = await currentApi.listContacts();
        setContactList(result);
      } catch (err) {
        setTimeout(() => {
          setErrorMessage('Sorry! a problem has occurred when getting contacts.');
        }, 0);
      }
    })();
  }, []);

  useEffect(() => {
    window.scrollTo(0, 0);
  }, [stepId]);

  useEffect(() => {
    const validationMessage = validateInputs();
    if (validationMessage) {
      return;
    }
    if (stepId === 'request-verify') {
      setShowVerifyModal(true);
    } else {
      setShowVerifyModal(false);
    }
    setCurrentStep(StepIndexes[stepId || 'request-detail']);
  }, [stepId]);

  useEffect(() => {
    if (errorMessage) window.scrollTo(0, 0);
  }, [errorMessage]);

  const handleRequestMoneyVerificationNext = () => {
    const data: InteracEtransferTransaction = {
      contactId: selectedContact,
      amount: mainInfo.amount,
      type: 'request',
    };
    setIsRequestingMoney(true);
    new Api(instance, accounts[0])
      .postInteracEtransferTransaction(data)
      .then(() => {
        setErrorMessage(undefined);
        setCurrentStep(2);
        navigateStep(PageIds.SentPageId);
      })
      .catch(() => setErrorMessage('Transfer failed.'))
      .finally(() => {
        setIsRequestingMoney(false);
        setShowVerifyModal(false);
      });
  };

  return (
    <div className="page-content">
      <RequestMoneyVerificationModal
        show={showVerifyModal}
        handleClose={handleRequestMoneyVerificationClose}
        handleNext={handleRequestMoneyVerificationNext}
        handleBack={handleRequestMoneyVerificationBack}
        isRequestingMoney={isRequestingMoney}
        transferDetails={getTransferDetails()}
      />
      <CommonPageContainer title="Request Money">
        {errorMessage && (
          <Alert variant="danger" className="zippy-btn rounded-0 text-dark py-2 my-2 px-5">
            <i />
            {errorMessage}
          </Alert>
        )}
        <Row>
          <Col xs={9}>
            <Row>
              <Col>
                <StepComponent
                  steps={2}
                  currentStep={currentStep}
                  setCurrentStep={(stepIndex: number) => {
                    const validationMessage = validateInputs();
                    if (validationMessage) {
                      setErrorMessage(validationMessage);
                      navigateStep(PageIds.DetailPageId);
                      return;
                    }
                    setCurrentStep(stepIndex);
                  }}
                  navigateSteps={navigateStep}
                />
              </Col>
            </Row>
            {(stepId === PageIds.DetailPageId || stepId === PageIds.VerifyPageId) && (
            <RequestDetails
              selectedContact={selectedContact}
              selectedAccount={selectedAccount}
              setContactToSend={setSelectedContact}
              setSelectedAccount={setSelectedAccount}
              setMainInfo={setMainInfo}
              mainInfo={mainInfo}
              accounts={accountsList || []}
              contacts={contactList || []}
              setErrorMessage={setErrorMessage}
              navigateStep={navigateStep}
              validateInputs={validateInputs}
            />
            )}
            {stepId === PageIds.SentPageId && (
            <RequestSent
              accounts={accountsList}
              contacts={contactList}
              selectedAccount={selectedAccount}
              selectedContact={selectedContact}
              mainInfo={mainInfo}
              setContactToSend={setSelectedContact}
              setSelectedAccount={setSelectedAccount}
              setMainInfo={setMainInfo}
              setCurrentStep={setCurrentStep}
              navigateStep={navigateStep}
            />
            )}
            <Divider />
            <Row>
              <Col>
                <b>Note:</b>
                <br />
                Your use of
                {' '}
                <i>Interac</i>
                {' '}
                e-Transfer is subject to the
                {' '}
                <i>Interac</i>
                {' '}
                e-Transfer Terms and conditions (PDF, 197 KB).
              </Col>
            </Row>
          </Col>
          <Col xs={3}>
            <div className="border p-2 rounded">
              <b>You can also:</b>
              <br />
              View upcoming bill payments and transfers &gt;
              <br />
              Stop a payment &gt;
              <br />
              Change your statement preferences &gt;
            </div>
          </Col>
        </Row>
      </CommonPageContainer>
    </div>
  );
}
