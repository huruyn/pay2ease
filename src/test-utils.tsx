import { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import './i18n/config';
import { store } from './app/store';

const AllTheProviders: FC = ({ children }) => (
  <Provider store={store}>
    <BrowserRouter>{children}</BrowserRouter>
  </Provider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as renderWithProviders };
