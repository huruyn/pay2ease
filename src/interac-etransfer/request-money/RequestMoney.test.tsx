import { BrowserRouter } from 'react-router-dom';
import { renderWithProviders, fireEvent } from '../../test-utils';
import RequestMoney from './RequestMoney';

const ReactTestRenderer = require('react-test-renderer');

describe('Request Money Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(<BrowserRouter><RequestMoney /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should click next step button', () => {
    const { container } = renderWithProviders(<RequestMoney />);
    fireEvent.click(container.querySelectorAll('div[role="button"]')[0]);
  });
});
