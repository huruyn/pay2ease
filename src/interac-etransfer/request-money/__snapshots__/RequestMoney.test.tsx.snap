// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Money Component matches the snapshot 1`] = `
<div
  className="page-content"
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <h3>
      Request Money
    </h3>
  </div>
  <div
    className="content-wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col-9"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <div
              className="d-flex align-items-center mt-3 mb-3"
            >
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                    "marginRight": 10,
                  }
                }
              >
                Steps: 
              </span>
              <div
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#C41F3E",
                    "border": "2px solid #C41F3E",
                    "borderRadius": 50,
                    "color": "white",
                    "display": "flex",
                    "height": 25,
                    "justifyContent": "center",
                    "width": 25,
                  }
                }
              >
                1
              </div>
              <div
                style={
                  Object {
                    "backgroundColor": "grey",
                    "height": 2,
                    "width": 15,
                  }
                }
              />
              <div
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                style={
                  Object {
                    "alignItems": "center",
                    "border": "2px solid grey",
                    "borderRadius": 50,
                    "display": "flex",
                    "height": 25,
                    "justifyContent": "center",
                    "width": 25,
                  }
                }
              >
                2
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <h4
                className="mt-4"
              >
                Request Money Details
              </h4>
            </div>
          </div>
          <div
            className="align-items-center mt-4 row"
          >
            <div
              className="col-3"
            >
              Request Money From:
            </div>
            <div
              className="col-6"
            >
              <select
                className="form-select"
                onChange={[Function]}
                value={0}
              >
                <option
                  value=""
                >
                  Select
                </option>
              </select>
            </div>
          </div>
          <div
            className="border-top my-3"
          />
          <div
            className="align-items-center row"
          >
            <div
              className="col-3"
            >
              Request Amount:
            </div>
            <div
              className="col-6"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="$"
                step=".01"
                type="number"
                value={0}
              />
            </div>
          </div>
          <div
            className="align-items-center mt-2 row"
          >
            <div
              className="col-3"
            />
            <div
              className="col-9"
            >
              You can request up to $10,000
              <br />
              Financial institutions may have different limits for using
               
              <i>
                Interac
              </i>
               
              e-Transfer. Confirm with your contact that they can send you the amount you're requesting.
            </div>
          </div>
          <div
            className="border-top my-3"
          />
          <div
            className="align-items-center row"
          >
            <div
              className="col-3"
            >
              Deposit To:
            </div>
            <div
              className="col-9"
            >
              <select
                className="form-select"
                onChange={[Function]}
                value={0}
              >
                <option>
                  Select
                </option>
              </select>
            </div>
          </div>
          <div
            className="align-items-center mt-2 row"
          >
            <div
              className="col-3"
            />
            <div
              className="col-9"
            >
              If your request is fulfilled, the money will be deposited automatically to your chosen account.
            </div>
          </div>
          <div
            className="border-top my-3"
          />
          <div
            className="align-items-center row"
          >
            <div
              className="col-3"
            >
              Invoice Due Date (Optional):
            </div>
            <div
              className="col-9"
            >
              <div
                className="row"
              >
                <div
                  className="col-auto"
                >
                  <select
                    className="form-select"
                  >
                    <option
                      value={1}
                    >
                      1
                    </option>
                    <option
                      value={2}
                    >
                      2
                    </option>
                    <option
                      value={3}
                    >
                      3
                    </option>
                    <option
                      value={4}
                    >
                      4
                    </option>
                    <option
                      value={5}
                    >
                      5
                    </option>
                    <option
                      value={6}
                    >
                      6
                    </option>
                    <option
                      value={7}
                    >
                      7
                    </option>
                    <option
                      value={8}
                    >
                      8
                    </option>
                    <option
                      value={9}
                    >
                      9
                    </option>
                    <option
                      value={10}
                    >
                      10
                    </option>
                    <option
                      value={11}
                    >
                      11
                    </option>
                    <option
                      value={12}
                    >
                      12
                    </option>
                  </select>
                </div>
                <div
                  className="col-auto"
                >
                  <select
                    className="form-select"
                  >
                    <option
                      value={1}
                    >
                      1
                    </option>
                    <option
                      value={2}
                    >
                      2
                    </option>
                    <option
                      value={3}
                    >
                      3
                    </option>
                    <option
                      value={4}
                    >
                      4
                    </option>
                    <option
                      value={5}
                    >
                      5
                    </option>
                    <option
                      value={6}
                    >
                      6
                    </option>
                    <option
                      value={7}
                    >
                      7
                    </option>
                    <option
                      value={8}
                    >
                      8
                    </option>
                    <option
                      value={9}
                    >
                      9
                    </option>
                    <option
                      value={10}
                    >
                      10
                    </option>
                    <option
                      value={11}
                    >
                      11
                    </option>
                    <option
                      value={12}
                    >
                      12
                    </option>
                    <option
                      value={13}
                    >
                      13
                    </option>
                    <option
                      value={14}
                    >
                      14
                    </option>
                    <option
                      value={15}
                    >
                      15
                    </option>
                    <option
                      value={16}
                    >
                      16
                    </option>
                    <option
                      value={17}
                    >
                      17
                    </option>
                    <option
                      value={18}
                    >
                      18
                    </option>
                    <option
                      value={19}
                    >
                      19
                    </option>
                    <option
                      value={20}
                    >
                      20
                    </option>
                    <option
                      value={21}
                    >
                      21
                    </option>
                    <option
                      value={22}
                    >
                      22
                    </option>
                    <option
                      value={23}
                    >
                      23
                    </option>
                    <option
                      value={24}
                    >
                      24
                    </option>
                    <option
                      value={25}
                    >
                      25
                    </option>
                    <option
                      value={26}
                    >
                      26
                    </option>
                    <option
                      value={27}
                    >
                      27
                    </option>
                    <option
                      value={28}
                    >
                      28
                    </option>
                    <option
                      value={29}
                    >
                      29
                    </option>
                    <option
                      value={30}
                    >
                      30
                    </option>
                    <option
                      value={31}
                    >
                      31
                    </option>
                  </select>
                </div>
                <div
                  className="col-auto"
                >
                  <select
                    className="form-select"
                  >
                    <option
                      value={2022}
                    >
                      2022
                    </option>
                    <option
                      value={2023}
                    >
                      2023
                    </option>
                    <option
                      value={2024}
                    >
                      2024
                    </option>
                    <option
                      value={2025}
                    >
                      2025
                    </option>
                    <option
                      value={2026}
                    >
                      2026
                    </option>
                  </select>
                </div>
              </div>
            </div>
          </div>
          <div
            className="border-top my-3"
          />
          <div
            className="align-items-center row"
          >
            <div
              className="col-3"
            >
              Invoice Number (Optional):
            </div>
            <div
              className="col-9"
            >
              <input
                className="form-control"
                onChange={[Function]}
                type="number"
              />
            </div>
          </div>
          <div
            className="border-top my-3"
          />
          <div
            className="align-items-center row"
          >
            <div
              className="col-3"
            >
              Message Or Reason For Request (Optional):
            </div>
            <div
              className="col-9"
            >
              <textarea
                className="form-control"
                onChange={[Function]}
              />
            </div>
          </div>
          <div
            className="border-top my-3"
          />
          <div
            className="align-items-center row"
          >
            <div
              className="col"
            >
              This is how you will appear in all emails to your
               
              <i>
                Interac
              </i>
               
              e-Transfer contacts. If incorrect,
               
              <a
                className="sc-bdvvtL bFOhwT"
                href="/interac-etransfer/edit-my-profile"
                onClick={[Function]}
              >
                edit your profile
              </a>
              .
            </div>
          </div>
          <div
            className="align-items-center mt-4 row"
          >
            <div
              className="col-3"
            >
              Your Email Nickname:
            </div>
            <div
              className="col-6"
            >
              
            </div>
          </div>
          <div
            className="border-top my-3"
          />
          <div
            className="align-items-center row"
          >
            <div
              className="col-3"
            >
              Your Email Address:
            </div>
            <div
              className="col-6"
            />
          </div>
          <div
            className="align-items-center mt-4 row"
          >
            <div
              className="col"
            >
              <div
                className="px-4 py-2"
                style={
                  Object {
                    "backgroundColor": "#606366",
                    "color": "#fff",
                  }
                }
              >
                <div
                  className="form-check"
                >
                  <input
                    checked={false}
                    className="form-check-input"
                    disabled={false}
                    id="confirm"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <label
                    className="form-check-label"
                    htmlFor="confirm"
                    title=""
                  >
                    I confirm that I have an existing relationship with this contact.
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            className="align-items-center mt-4 row"
          >
            <div
              className="col"
            >
              <b>
                Note: You may be charged a fee when your request for money is completed. Fee may not apply depending on account terms.
                 
                <u>
                  Review fee details
                </u>
                .
              </b>
            </div>
          </div>
          <div
            className="border-top my-3"
          />
          <div
            className="row"
          >
            <div
              className="text-end col"
            >
              <button
                className="zippy-btn btn btn-primary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Send Request
              </button>
            </div>
          </div>
        </div>
        <div
          className="border-top my-3"
        />
        <div
          className="row"
        >
          <div
            className="col"
          >
            <b>
              Note:
            </b>
            <br />
            Your use of
             
            <i>
              Interac
            </i>
             
            e-Transfer is subject to the
             
            <i>
              Interac
            </i>
             
            e-Transfer Terms and conditions (PDF, 197 KB).
          </div>
        </div>
      </div>
      <div
        className="col-3"
      >
        <div
          className="border p-2 rounded"
        >
          <b>
            You can also:
          </b>
          <br />
          View upcoming bill payments and transfers &gt;
          <br />
          Stop a payment &gt;
          <br />
          Change your statement preferences &gt;
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer-content-wrapper"
  >
    <div
      className="quick-links container"
    >
      <div
        className="row"
      >
        <h3
          className="col-sm-12"
        >
          Quick Links
        </h3>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-2"
        >
          <ul>
            <li>
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li>
              Please Review Our Legal Stuff by visiting our
               
              <a
                className="text-decoration-none"
                href="/legal"
                onClick={[Function]}
              >
                Privacy Policy | TOU | EULA | CardHolder T&C Pages
              </a>
            </li>
          </ul>
        </div>
        <div
          className="col-sm-2"
        >
          8465 W Sahara Ave. Las Vegas, NEVADA
          <ul>
            <li>
              <a
                href="tel:+17026231876"
              >
                1.702.623.1876
              </a>
            </li>
            <li>
              <a
                href="mailto:contact@zippy.cash"
              >
                contact@zippy.cash
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 col-sm-12 col-xs-12 text-left"
        >
          © 2021 Zippy Cash LLC  All rights reserved.
          <br />
          <br />
          The Program Manager for the Zippy Cash® Prepaid Visa Debit Card is PayQuicker, LLC
          <br />
          Zippy Cash® is a registered trademark of Zippy Cash, LLC
          <br />
          The Zippy Cash Prepaid Visa Debit Card is issued by Metropolitan Commercial Bank (Member FDIC) pursuant to a license from Visa U.S.A. Inc. “Metropolitan Commercial Bank” and “Metropolitan” are registered trademarks of Metropolitan Commercial Bank ©2014. BY USING THIS CARD YOU AGREE TO THE TERMS AND CONDITIONS OF THE CARDHOLDER AGREEMENT AND FEE SCHEDULE, IF ANY. All trademarks and service marks are the property of their respective owners.
          <br />
          In compliance with the USA PATRIOT Act* and other laws, anyone who is a US resident over 18 with a valid Social Security number and a verifiable physical US address (no PO boxes or APO addresses) can apply for an Zippy Cash Card. This information must be provided, along with other information pertinent to receiving approval, when applying for a Zippy Cash Card. Currently, we do not issue the Zippy Cash Card to residents of Puerto Rico, Guam, the Virgin Islands, American Samoa, or other US territories.
          <br />
          *Important information for opening a card account: To help the federal government fight the funding of terrorism and money laundering activities, the USA PATRIOT Act requires all financial institutions and their third parties to obtain, verify and record information that identifies each person who opens a card account. What this means for you: When you open a card account, we will ask for your name, address, date of birth, social security number or country identification number, and other information that will allows us to identify you. We may also request a copy of your driver’s license or other documentation bearing your photo as verification of your identity.
          <br />
          Apple and the Apple logo are trademarks of Apple Inc., registered in the U.S. and other countries. App Store is a service mark of Apple Inc. Google Play is a trademark of Google Inc. Data rates may apply. Samsung Pay is a trademark of Samsung Inc.
        </div>
      </div>
    </div>
    <footer
      className="container mb-3"
    >
      <div
        className="d-none d-md-block"
      >
        <div
          className="d-flex justify-content-between border-bottom"
        />
        <div
          className="d-flex justify-content-between border-top"
        >
          <div
            className=""
          >
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/personal-profile"
              >
                dev docs
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Lore
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Ipsum
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Nullam Atoidio
              </a>
            </div>
          </div>
          <div
            className="copyright mt-1"
          >
            <span>
              © 2021 Lorem ipsum dolor sit amet, consectetuer
            </span>
          </div>
        </div>
      </div>
      <div
        className="d-block d-md-none"
      >
        <div
          className="d-flex border-bottom"
        >
          <div
            className="d-flex flex-column me-2"
          >
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/personal-profile"
              >
                dev docs
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Lore
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Ipsum
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Nullam Atoidio
              </a>
            </div>
          </div>
        </div>
        <div
          className="d-flex justify-content-end border-top"
        >
          <div
            className="copyright mt-2 text-end small"
          >
            <span>
              © 2021 Lorem ipsum dolor sit amet, consectetuer
            </span>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
